### PR TITLE
Add package to store POCs and pull in experimental table

### PR DESCRIPTION
**Pull request by fschade**
## Description
Add a package to keep track and try out POCs inside web.
It also adds a first experimental table component which adds the ability to group rows.

## Related Issue
- Needs https://github.com/owncloud/owncloud-design-system/pull/2048
- Fixes https://github.com/owncloud/owncloud-design-system/pull/1838

## Motivation and Context
be able to showcase certain experimental features

## How Has This Been Tested?
- local installation

## Types of changes
- [X] New Package

## Checklist:
- [X] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added

## Open tasks:

- [ ] add tests
- [ ] release and link ODS
- [ ] feedback from @elizavetaRa & @diocas

### CODE REVIEW

**Comment by fschade** [comment]
@diocas & @elizavetaRa 
can you please have a look if this fit your needs, you still have to manually import the experimental table,
but now web is able to create such components on it's own and does not need a custom ODS version anymore.

To do so please import and add the custom table component in the individual component:
```
...
import OcTable from 'web-pocs/src/components/OcTable/OcTable.vue'

export default {
  components: {
    OcTable
  },
  ...
}
```

please let me know if it's ok for you to do it manually,
as an alternative we could try to find a way to do it at build time... but then we have to discuss how we pull in the custom props which do not exist on the og table.

**Comment by diocas** [comment]
Our objective (which I think is aligned with oC management) is to drop forks on our side. I know we still need to compile ourselves (since we want some extra flags), but this solution means maintaining code (although small, so maybe acceptable for now).
But what I do not understand is the reluctance in making this table available to everyone. Before I though you didn't had it for lack of time to implement it, but if you implemented it, then what's the issue? Is it only the pagination (which you almost don't show now due to larger 'pages')?

**Comment by diocas** [comment]
Also, is this implementation completely separate from all the normal tables? Since we might become the sole users of it, will its compatibility and functionality be assured for future updates?

**Comment by kulmann** [comment]
@diocas this PR is the minimal effort to get your code upstream (to reduce patch complexity on your side) - we can't afford to spend more time with it at the moment. The PR goes to web because we don't want to increase code complexity for the `OcTable` component and we don't want to expose it as public API in ODS.  The code needs improvement (e.g. css classes from the ODS docs being used is a no-go, grouping selection doesn't belong in the table component, etc). Happy to iterate on this in a few weeks. For the time being I hope that this helps to keep your fork cleaner.

**Comment by diocas** [comment]
But can you clarify what are the future plans? 
Will this be available in oCIS web upstream's sharing views as well? Will you, once you have more time, try to deduplicate some of this code? While we're not there, will you ensure it's properly kept in sync, functionality wise, with OcTable?

**Comment by fschade** [comment]
@diocas the plan is for sure to pull in the grouping into core. Right now too many topics are on the table and we though that could be a quick and handy way to support you in the meantime.

so if you’re fine with that and it works for you as a temporary solution please let me know. Then I cleanup ods leftovers and create some basic test for it.

**Comment by diocas** [comment]
I think it's better than the current situation, so let's do it for now.
What cleanups do you need to do?
I've tested the table yesterday and there were things broken (I've asked @elizavetaRa to have a look and give feedback.. It might be that I've just messed up when adding it in the code)

**Comment by fschade** [comment]
@diocas, @elizavetaRa i rebased master, no local ODS linking needed anymore.

**Comment by fschade** [comment]
@elizavetaRa have you looked into it in the meanwhile? 
@diocas wanna get rid of ods jsdocs and add some testing.

**Comment by elizavetaRa** [comment]
@fschade We are using it locally already, I found some bugs, so I am doing some fixing & cleanup here [#6748](https://github.com/owncloud/web/pull/6748)

**Comment by fschade** [comment]
@elizavetaRa thank you for the feedback, please let me know when your pr is ready for review :)

**Comment by diocas** [comment]
Hi @fschade, any update?

**Comment by fschade** [comment]
> Hi @fschade, any update?

Today is my first day after vacation, i'll have a look.

**Comment by diocas** [comment]
@fschade since we still need to have a commit on our side adding this table, would you mind if we made it enabled via capability instead? (or, I don't know if possible/easy to do, maybe a compile flag?)

**Comment by fschade** [comment]
@diocas ill discuss this in a sec which way to go...
the bigger problem i have right now is that we're in the ocis beta phase right now and i think theres no time this week to finalize it.

Is it enough for you if we add it next week?

**Comment by fschade** [comment]
@diocas is it ok if we use a config option for that?
https://github.com/owncloud/web/blob/95edac2b147591febfa1b757b2073a49f54a7e68/config/config.json.sample-ocis#L32

**Comment by diocas** [comment]
Yes, I wanted to say config option. No problem in doing it next week, but I'll see if we can push a PR with this option.

**Comment by kulmann** [comment]
@fschade could you rebase this PR and incorporate the config option `options.cernFeatures`?

**Comment by diocas** [comment]
@kulmann, @elizavetaRa opened a PR (#7005) with an `enableAdvancedTable` option and some fixes (the rebase is still necessary on this branch, though).

**Comment by diocas** [comment]
Hi @fschade , when can you merge Lisa's changes and merge this as well?
Btw, is this table up to date with the latest improvements to the files listing?

**Comment by diocas** [comment]
We've been keeping this code up to date. You can take if from this commit: https://github.com/cernbox/web/commit/80af010d

File: yarn.lock [hunk 1]
```diff
@@ -13801,6 +13801,12 @@ __metadata:
   languageName: unknown
   linkType: soft
 
+"web-pocs@workspace:packages/web-pocs":
+  version: 0.0.0-use.local
+  resolution: "web-pocs@workspace:packages/web-pocs"
+  languageName: unknown
+  linkType: soft
+
```
Comment (dschmidt) [inline yarn.lock:13809]: Excuse me, wir haben 2022?!

Where does the yarn.lock file come from? I don't see it in master and we shouldn't reads it

**Review by dschmidt — CHANGES_REQUESTED** [review]

**Comment by diocas** [comment]
Rebased with latest master from here: https://github.com/cernbox/web/commit/2ccaafa8d4ffa10f3ff6d869268beb0183791a4e

It also configures web to use it, with a configurable option (maybe with that we could put it out of PoCs and into the real thing?).

**Comment by pascalwengerter** [comment]
@kulmann I think this PR can be closed as superseded via #8443 ?

**Comment by kulmann** [comment]
as @pascalwengerter correctly pointed out:
closing as superseded via https://github.com/owncloud/web/pull/8443

**Pull request closed without merging**